gc/base/standard/ConcurrentGC.cpp

```diff
@@ -1582,7 +1582,7 @@ MM_ConcurrentGC::signalThreadsToActivateWriteBarrier(MM_EnvironmentBase *env)
 		/* We may or may not have exclusive access but another thread may have beat us to it and
 		 * prepared the threads or even collected.
 		 */
-		if (env->acquireExclusiveVMAccessForGC(this, true, false)) {
+		if (env->acquireExclusiveVMAccessForGC(this, true, _extensions->usingSATBBarrier())) {
```

Comment:
you can add a comment like:
SATB marks all newly allocated objectes during active concurrent cycle. Since it's done on TLH granularity we have to flush the current ones and start creating new ones, once the cycle starts.